### ISSUE TITLE
The index of a ngFor is not recongnized as a number again

### ISSUE THREAD

**Issue by kartercs** [angular/vscode-ng-language-service#670]
Following this issue [#514 ](https://github.com/angular/vscode-ng-language-service/issues/514)
It looks like the bug came back on version 0.900.15 (wich i'm currently using).

I have this really simple piece of code : 

```html
<div fxLayout="row" fxLayoutGap="20px grid" fxLayoutAlign="space-evenly center" fxFlex style="text-align: center;">
    <div *ngFor="let table of title; let i = index" style='cursor: pointer;'>
        <div *ngIf='table.length == 0' style='text-align: center;'> Aucune table n'est disponnible dans cette base de donnée </div>
        <div *ngIf='table.length != 0'> <img src='assets/images/table.png' width="50px"><br /> {{table}} </div>
        <div *ngIf="i == 7" class='break'></div>
    </div>
</div>
```

And i have the following error from the langage service :
" Expected the operants to be of similar type or any ng(0)" on this part "*ngIf="i == 7" event thought it's for sure a number.

When i disable the angular Language Service i don't have the error anymore.

Thanks in advance  for your help :)

_edit ayazhafiz_: code formatting

**Comment by ayazhafiz** [comment]
Can you list the console output of the Angular language service (View > Output > Angular Language Service dropdown)?

**Comment by ivanwonder** [comment]
@kartercs please check if you define the `index` in your component context.

**Comment by ayazhafiz** [comment]
@ivanwonder I think `index` should be exported by the NgFor directive's template context

**Comment by ivanwonder** [comment]
@ayazhafiz because the LS infer the `index` type to be `index` in the component context.
https://github.com/angular/angular/pull/35982#discussion_r390107111

**Comment by ayazhafiz** [comment]
Fixed by @ivanwonder's PR: angular/angular#35982

**Comment by kartercs** [comment]
Thanks a lot to both of you ! I'm looking forward to using the language service again :)
